kubectl-fdb/cmd/profile_analyzer.go

```diff
@@ -0,0 +1,198 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			config, err := o.configFlags.ToRESTConfig()
+			if err != nil {
+				return err
+			}
+			clientSet, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return err
+			}
+			dynamicConfig, err := dynamic.NewForConfig(config)
+			if err != nil {
+				return err
+			}
+			clusterName, err := cmd.Flags().GetString("fdb-cluster")
+			if err != nil {
+				return err
+			}
+			startTime, err := cmd.Flags().GetString("start-time")
+			if err != nil {
+				return err
+			}
+			endTime, err := cmd.Flags().GetString("end-time")
+			if err != nil {
+				return err
+			}
+			topRequests, err := cmd.Flags().GetInt("top-requests")
+			if err != nil {
+				return err
+			}
+			templateName, err := cmd.Flags().GetString("template-name")
+			if err != nil {
+				return err
+			}
+
+			namespace, err := getNamespace(*o.configFlags.Namespace)
+			if err != nil {
+				return err
+			}
+
+			return runProfileAnalyzer(clientSet, dynamicConfig, namespace, clusterName, startTime, endTime, topRequests, templateName)
+		},
+		Example: `
+# Run the profiler for cluster-1. We require --cluster option explicitly because analyze commands take lot many arguments.
+kubectl fdb analyze-profile -c cluster-1 --start-time "01:01 20/07/2022 BST" --end-time "01:30 20/07/2022 BST" --top-requests 100
+`,
+	}
+	cmd.SetOut(o.Out)
+	cmd.SetErr(o.ErrOut)
+	cmd.SetIn(o.In)
+
+	cmd.Flags().StringP("fdb-cluster", "c", "", "cluster name for running hot shard tool.")
+	cmd.Flags().String("start-time", "", "start time for the analyzing transaction '01:30 30/07/2022 BST'")
+	cmd.Flags().String("end-time", "", "end time for analyzing the transaction '02:30 30/07/2022 BST'")
+	cmd.Flags().String("template-name", "", "Name of the Job template")
+	cmd.Flags().Int("top-requests", 100, "")
+	err := cmd.MarkFlagRequired("fdb-cluster")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	o.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func runProfileAnalyzer(kubeClient *kubernetes.Clientset, dynamicConfig dynamic.Interface, namespace string, clusterName string, startTime string, endTime string, topRequests int, templateName string) error {
+	pc := profileConfig{
+		Namespace:   namespace,
+		ClusterName: clusterName,
+		JobName:     clusterName + "-hot-shard-tool",
+		CommandArgs: " -C " + " /var/dynamic-conf/fdb.cluster" + " -s \"" + startTime + "\"" + " -e \"" + endTime + "\"" + " --filter-get-range " + " --top-requests " + strconv.Itoa(topRequests),
+	}
+	t, err := template.ParseFiles(templateName)
+	if err != nil {
+		return err
+	}
+	buf := bytes.Buffer{}
+	err = t.Execute(&buf, pc)
+	if err != nil {
+		return err
+	}
+	decoder := yamlutil.NewYAMLOrJSONDecoder(&buf, 100000)
+
+	for {
+		var rawObj runtime.RawExtension
+		err := decoder.Decode(&rawObj)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		obj, gvk, err := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).
```

Comment:
No need for this additional complexity if we only expect a Job we can directly decode it as a Job.